core/runtime/execute_engine.cpp

```diff
@@ -44,7 +44,7 @@ bool is_switch_required(const RTDevice& curr_device, const RTDevice& engine_devi
 }
 
 RTDevice select_rt_device(const RTDevice& engine_device) {
-  auto new_target_device_opt = get_most_compatible_device(engine_device);
+  auto new_target_device_opt = get_most_compatible_device(engine_device, /*prioritize_current_device=*/true);
```

Comment:
Only in the _runtime_ session (`select_rt_device`), do we want to prioritize the current device over the target device. This is not the case in the other call to `get_most_compatible_device`, which occurs in the `TRTEngine` constructor.